subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/CacheLayout.java

```diff
@@ -15,49 +15,72 @@
  */
 package org.gradle.api.internal.artifacts.ivyservice;
 
-import org.gradle.util.VersionNumber;
+import org.gradle.cache.internal.CacheVersion;
+import org.gradle.cache.internal.CacheVersionMapping;
 
 import java.io.File;
 
+import static org.gradle.cache.internal.CacheVersionMapping.introducedIn;
+
 public enum CacheLayout {
-    ROOT(null, "modules", 2),
-    FILE_STORE(ROOT, "files", 1),
-    META_DATA(ROOT, "metadata", 58),
-    RESOURCES(ROOT, "resources", 1),
-    TRANSFORMS(null, "transforms", 1),
-    TRANSFORMS_META_DATA(TRANSFORMS, "metadata", 1),
-    TRANSFORMS_STORE(TRANSFORMS, "files", 1);
-
-    // If you update the META_DATA version, also update DefaultGradleDistribution.getArtifactCacheLayoutVersion() (which is the historical record)
+
+    ROOT(null, "modules", introducedIn("1.9-rc-1").incrementedIn("1.9-rc-2")),
+
     // If you update FILE_STORE, you may also need to update LocallyAvailableResourceFinderFactory
+    FILE_STORE(ROOT, "files", introducedIn("1.9-rc-1")),
+
+    META_DATA(ROOT, "metadata",
+        // 3-5, 7-11, 18-20, 22, 25-30, 32-35, 37-47, 49, 50, 52, 54, 55, 57 were never used in a release
```

Comment:
I don't think this comment needs to be specific. It can just say "versions may be skipped if they were never released"

Reply:
Ok, I'll change it.